Orange/widgets/visualize/owscatterplot.py

```diff
@@ -428,19 +390,13 @@ def set_attr(self, attr_x, attr_y):
             self.attr_changed()
 
     def attr_changed(self):
-        self.cb_reg_line.setEnabled(self.can_draw_regresssion_line())
         self.setup_plot()
         self.commit()
 
     def setup_plot(self):
         super().setup_plot()
-        for axis, var in (("bottom", self.attr_x), ("left", self.attr_y)):
-            self.graph.set_axis_title(axis, var)
-            if var and var.is_discrete:
-                self.graph.set_axis_labels(axis,
-                                           get_variable_values_sorted(var))
-            else:
-                self.graph.set_axis_labels(axis, None)
+        self.graph.set_axis_title("bottom", self.attr_x)
+        self.graph.set_axis_title("left", self.attr_y)
```

Comment:
Shouldn't this be implemented on the graph?
It is indeed an old issue, but the axis titles are not cleared when data is removed.

Reply:
Yes, let's do it in another PR